liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -196,6 +227,38 @@ unoptimiseDynFlags df = updOptLevel 0 df
     , ghcMode      = CompManager
     }
 
+--------------------------------------------------------------------------------
+-- | Parsing phase -------------------------------------------------------------
+--------------------------------------------------------------------------------
+
+-- | We hook at this stage of the pipeline to store the parsed module
+-- for later processing during typechecking:
+--
+-- * Comments are preserved during parsing because we turn on
+--   'Opt_KeepRawTokenStream'. So we can get the /LH/ specs from there.
+--
+-- * The typechecker hook needs to do desugaring itself (see the
+--   comment on 'typecheckHook') which requires access to a full
+--   'TypecheckedModule'. Annoyingly, the typechecker hook is given
+--   only a 'TcGblEnv', so we have to redo typechecking there. And
+--   the input to that needs to be a 'ParsedModule'.
+--
+parsedHook :: [CommandLineOption] -> ModSummary -> ParsedResult -> Hsc ParsedResult
+parsedHook _ ms parsedResult = do
```

Comment:
Should the breadcrumb be inserted only if the module is not skipped in `liquidPlugin`? Otherwise, the AST is going to be retained until the call to GHC finishes.